side note, I appear to have broken my dev env:

```
[deads@deads-02 kubernetes]$ make WHAT=cmd/kube-apiserver
stat /home/deads/workspaces/kubernetes/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/third_party/etcd-v3.3.15-linux-amd64/Documentation/README.md: no such file or directory
14d13
< API rule violation: list_type_missing,k8s.io/apimachinery/pkg/apis/meta/v1,List,Items
18d16
< API rule violation: list_type_missing,k8s.io/apimachinery/pkg/apis/meta/v1,PartialObjectMetadataList,Items
29,31d26
< API rule violation: list_type_missing,k8s.io/code-generator/_examples/apiserver/apis/example/v1,TestTypeList,Items
< API rule violation: list_type_missing,k8s.io/code-generator/_examples/apiserver/apis/example2/v1,TestTypeList,Items
< API rule violation: list_type_missing,k8s.io/code-generator/_examples/apiserver/apis/example3.io/v1,TestTypeList,Items
ERROR: 
	 CODEGEN API rule check failed. Reported violations differ from known violations. Please read api/api-rules/README.md to resolve the failure in api/api-rules/codegen_violation_exceptions.list. 

make[1]: *** [Makefile.generated_files:504: staging/src/k8s.io/code-generator/_examples/apiserver/openapi/zz_generated.openapi.go] Error 1
make: *** [Makefile:544: generated_files] Error 2

```

--- a/cmd/kube-apiserver/app/options/BUILD
+++ b/cmd/kube-apiserver/app/options/BUILD
@@ -28,6 +28,7 @@ go_library(
         "//staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver:go_default_library",
         "//staging/src/k8s.io/apimachinery/pkg/util/net:go_default_library",
         "//staging/src/k8s.io/apiserver/pkg/admission:go_default_library",
+        "//staging/src/k8s.io/apiserver/pkg/features:go_default_library",
         "//staging/src/k8s.io/apiserver/pkg/server/options:go_default_library",
         "//staging/src/k8s.io/apiserver/pkg/storage/storagebackend:go_default_library",
         "//staging/src/k8s.io/apiserver/pkg/util/feature:go_default_library",

--- a/cmd/kube-apiserver/app/options/validation.go
+++ b/cmd/kube-apiserver/app/options/validation.go
@@ -23,6 +23,7 @@ import (
 	"strings"
 
 	apiextensionsapiserver "k8s.io/apiextensions-apiserver/pkg/apiserver"
+	genericfeatures "k8s.io/apiserver/pkg/features"
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
 	"k8s.io/component-base/metrics"
 	aggregatorscheme "k8s.io/kube-aggregator/pkg/apiserver/scheme"
@@ -127,6 +128,25 @@ func validateTokenRequest(options *ServerRunOptions) []error {
 	return errs
 }
 
+func validateAPIPriorityAndFairness(options *ServerRunOptions) []error {
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.APIPriorityAndFairness) && options.GenericServerRunOptions.EnablePriorityAndFairness {
+		// we know we need the alpha API enabled.  There are only a few ways to turn it on
+		enabledAPIString := options.APIEnablement.RuntimeConfig.String()
+		switch {
+		case strings.Contains(enabledAPIString, "api/all=true"):
+			return nil
+		case strings.Contains(enabledAPIString, "api/alpha=true"):
+			return nil
+		case strings.Contains(enabledAPIString, "flowcontrol.apiserver.k8s.io/v1alpha1=true"):
+			return nil
+		default:
+			return []error{fmt.Errorf("enabling APIPriorityAndFairness requires --runtime-confg=flowcontrol.apiserver.k8s.io/v1alpha1=true to enable the required API")}
+		}
+	}
+
+	return []error{}
+}
+
 // Validate checks ServerRunOptions and return a slice of found errs.
 func (s *ServerRunOptions) Validate() []error {
 	var errs []error
@@ -136,6 +156,7 @@ func (s *ServerRunOptions) Validate() []error {
 	errs = append(errs, s.Etcd.Validate()...)
 	errs = append(errs, validateClusterIPFlags(s)...)
 	errs = append(errs, validateServiceNodePort(s)...)
+	errs = append(errs, validateAPIPriorityAndFairness(s)...)
 	errs = append(errs, s.SecureServing.Validate()...)
 	errs = append(errs, s.Authentication.Validate()...)
 	errs = append(errs, s.Authorization.Validate()...)

